samcli/lib/init/__init__.py

```diff
@@ -121,9 +129,19 @@ def generate_project(
 
     _apply_tracing(tracing, output_dir, name)
 
+    _enable_application_insights(application_insights, output_dir, name)
+
 
 def _apply_tracing(tracing: bool, output_dir: str, name: str) -> None:
     if tracing:
         template_file_path = f"{output_dir}/{name}/template.yaml"
         template_modifier = XRayTracingTemplateModifier(template_file_path)
         template_modifier.modify_template()
+
+
+def _enable_application_insights(application_insights: bool, output_dir: str, name: str) -> None:
```

Comment:
So if I understand correctly, this enables Application Insights by changing the input template. This means that unless someone is using SAM CLI (e.g. deploying directly with `aws cloudformation deploy`), it won't include the extra resources and might cause them to get deleted.

Should this be done in the [SAM transform](https://github.com/aws/serverless-application-model/) instead?

Reply:
Ah on second glance I might've misunderstood. I assume this only runs once during `sam init` and is stored in the final template?

If that's the case, this could potentially also live as part of the [`init` templates](https://github.com/aws/aws-sam-cli-app-templates). 🤔

Reply:
> I assume this only runs once during sam init and is stored in the final template?

Instead of the final template, the snippet (App Insights resources) is added into the working template. 

This change is similar to the [X-Ray Tracing change](https://github.com/aws/aws-sam-cli/pull/3817). We considered options like the current choice (adding an option in the init flow to add or not add the resources) and adding the snippet in init templates. We decided to go with the current approach because it's not flexible to allow enable/no-enable in the init templates repo. Also, if we made the change in init templates instead, we would have to modify each template to have one version with the added resources and one version without.